src/managed/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.Netstandard2_0.cs

```diff
@@ -0,0 +1,628 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.DependencyModel
+{
+    public partial class DependencyContextJsonReader : IDependencyContextReader
+    {
+        public DependencyContext Read(Stream stream)
+        {
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+
+            ArraySegment<byte> drained = ReadToEnd(stream);
+
+            DependencyContext dependencyContext;
+            try
+            {
+                dependencyContext = ReadCore(drained.AsSpan(0, drained.Count));
+            }
+            catch
+            {
+                // Holds document content, clear it before returning it.
+                drained.AsSpan(0, drained.Count).Clear();
+                ArrayPool<byte>.Shared.Return(drained.Array);
+                throw;
+            }
+            finally
+            {
+                // Holds document content, clear it before returning it.
+                drained.AsSpan(0, drained.Count).Clear();
+                ArrayPool<byte>.Shared.Return(drained.Array);
+            }
+            return dependencyContext;
+        }
+
+        private static bool IsTokenTypeProperty(JsonTokenType tokenType)
+            => tokenType == JsonTokenType.PropertyName;
+
+        private DependencyContext ReadCore(ReadOnlySpan<byte> jsonData)
```

Comment:
Can this be static?  It's hard for me to see whether this is using any instance state.

Reply:
No, it can't be static. Many of the methods it calls reference `Pool(string s)` which uses the `_stringPool`.
```C#
private readonly IDictionary<string, string> _stringPool = new Dictionary<string, string>();
```

A few of the other methods can be made static since they don't call the Pool instance method. Should I update those? For example:
```C#
private static List<string> ReadPropertyNames(ref Utf8JsonReader reader);
private static List<RuntimeFallbacks> ReadRuntimes(ref Utf8JsonReader reader);
```